### ISSUE TITLE
PDF generation with latexpdf: adding a preamble in conf.py doesn't seem to work on readthedocs.io

### ISSUE THREAD

**Issue by gerv** [rtfd/readthedocs.org#2333]
Steps to reproduce:
- Download http://readthedocs.org/projects/bugzilla/downloads/pdf/latest/
- Note that the TOC only has a single level
- View http://git.mozilla.org/?p=bugzilla/bugzilla.git;a=blob;f=docs/en/rst/conf.py
- Note the line: 'preamble': '\setcounter{tocdepth}{4}',
- Be surprised.

This works when building on my local Ubuntu machine - I can control the tocdepth between 0 and whatever, and the output responds. So that should mean the command and syntax are correct, and leads me to suspect there's something different about the live RTD website builders which make this not work for some reason.

See related issue #1997 , which I filed because I thought a different conf.py command controlled this. But I've now tested with proper latexpdf building and it still doesn't work.

Gerv

**Comment by davidchall** [comment]
I also experienced this issue, but resolved it by requiring a later version of sphinx in my requirements.txt file. Perhaps this can help you too?

**Comment by humitos** [comment]
@gerv what's the difference between this issue and the one you listed, #1997

**Comment by humitos** [comment]
This issue is something that called my attention.

Running my local RTD instance, importing manually the bugzilla project and building the PDF I can see the TOC properly with the correct deep.

![captura de pantalla_2017-03-09_18-49-09](https://cloud.githubusercontent.com/assets/244656/23775903/1af625b0-04f9-11e7-9127-1069e99a557b.png)

I'm using the `readthedocs/build:2.0` image, but... not the `2.0` that was released yesterday, the older one. Maybe we loose a dependency in the middle? I'm downloading the new one and I will test again with that one.

Anyway, his last build was http://readthedocs.org/projects/bugzilla/builds/5088877/ that it's one week old. I re-trigger a new build of `latest` to see what happen in production and the TOC depth is not correct also.

I found this in the logs:

```
** (sphinx) defining (legacy) text style macros without \sphinx prefix
** if clashes with packages, set latex_keep_old_macro_names=False in conf.py

....

******** ERROR !! PLEASE UPDATE titlesec.sty !!********
******** THIS VERSION SWALLOWS SECTION NUMBERS.********.
```

that seems to be related to the latest upgrade to Sphinx 1.5.3, https://github.com/rtfd/readthedocs.org/issues/2706#issuecomment-285267252
